Warning! Could not find the package with 'Update' requirements for your system 'linux_x86_64'

### The problem

Can not compile after upgrading to  2021.10.0
![image](https://user-images.githubusercontent.com/65124087/138216489-c9899ed8-0cbb-41d4-ba6f-243af936afc9.png)


### Which version of ESPHome has the issue?

2021.10.0

### What type of installation are you using?

Docker

### Which version of Home Assistant has the issue?

core-2021.10.6

### What platform are you using?

ESP8266

### Board

_No response_

### Component causing the issue

undefined reference to `esphome::light::ESPColorView:

### Example YAML snippet

```yaml
substitutions:
  devicename: espblb1_rroom
  

esphome:
  name: $devicename
  platform: ESP8266
  board: esp_wroom_02

packages:
  common: !include common/common.yaml

output:
  - platform: esp8266_pwm
    id: output_red
    pin: GPIO4
  - platform: esp8266_pwm
    id: output_green
    pin: GPIO12
  - platform: esp8266_pwm
    id: output_blue
    pin: GPIO14
  - platform: esp8266_pwm
    id: output_white
    pin: GPIO5

e131:
  method: unicast

light:
  - platform: rgbw
    internal: true
    id: l1
    red: output_red
    green: output_green
    blue: output_blue
    white: output_white

  - platform: partition
    name: $devicename
    id: $devicename
    segments:
      - id: l1
        from: 0
        to: 0      
    effects:
      - e131:
          universe: 1
          channels: RGBW    
      - random:
      - flicker:
```


### Anything in the logs that might be useful for us?

_No response_

### Additional information

_No response_

@Marlor I meant I can't reproduce the compile error "undefined reference to ..." with the config. The title of this issue is a bit misleading, the Update warning is just that, a warning. iirc the warning is printed because we need to add the Update library from arduino core, _but_ PIO somehow can't find it (but we still need to define it so that the includes work) - in short: it's complicated

Re the more than one found: please try with 2021.10, most of those should be gone now

--- a/esphome/components/light/addressable_light.h
+++ b/esphome/components/light/addressable_light.h
@@ -22,6 +22,12 @@ using ESPColor ESPDEPRECATED("esphome::light::ESPColor is deprecated, use esphom
 /// Convert the color information from a `LightColorValues` object to a `Color` object (does not apply brightness).
 Color color_from_light_color_values(LightColorValues val);
 
+/// Use a custom state class for addressable lights, to allow type system to discriminate between addressable and
+/// non-addressable lights.
+class AddressableLightState : public LightState {
+  using LightState::LightState;
+};
+
 class AddressableLight : public LightOutput, public Component {
  public:
   virtual int32_t size() const = 0;

--- a/esphome/components/light/types.py
+++ b/esphome/components/light/types.py
@@ -4,10 +4,9 @@ from esphome import automation
 # Base
 light_ns = cg.esphome_ns.namespace("light")
 LightState = light_ns.class_("LightState", cg.EntityBase, cg.Component)
-# Fake class for addressable lights
-AddressableLightState = light_ns.class_("LightState", LightState)
+AddressableLightState = light_ns.class_("AddressableLightState", LightState)
 LightOutput = light_ns.class_("LightOutput")
-AddressableLight = light_ns.class_("AddressableLight", cg.Component)
+AddressableLight = light_ns.class_("AddressableLight", LightOutput, cg.Component)
 AddressableLightRef = AddressableLight.operator("ref")
 
 Color = cg.esphome_ns.class_("Color")

